### ISSUE TITLE
v-annotate.pl bug: cdsstopp alert causes script to fail if early stop position exceeds CDS model length

### ISSUE THREAD

**Issue by nawrockie** [nawrockie/dnaorg_annotate_genomes.pl#84]
If a CDS has multiple protein sequences in its blastx library, and some of those are longer than the reference length for the CDS, then it is possible to have an early stop in the blastx output occur at a position that exceeds the reference CDS length. The code expects this is impossible and it eventually causes the program to fail with an error message like:

```
vdr_CoordsRelativeSegmentToAbsolute, relative coords segment has a position 240 that exceeds absolute coords (6247..6447:+) length: 201 
```

The example that causes this error was reported by a user but I decided it was too difficult/inconvenient to add an example to the testsuite, so I will just fix it without adding an example to the testsuite.

**Comment by nawrockie** [comment]
Fix is to explicitly check if the early stop position exceeds the reference CDS position and if so to handle it in a special way, but not trying to calculate the model position of the early stop. 

Typically this type of `cdsstopp` alert in a protein that is not the default (model) protein sequence would have an `alert detail` field in the `.alt` output file like this:

```
stop codon in protein-based alignment [mdl-coords_wrt:<accn>.1:6247..6444:+]
```
and with `mdl_coords` field of `196..198:+` 

The situation that expresses this bug makes it impossible to determine the model coordinates for the early stop, and so I decided to make the `mdl_coords` field for this situation to be blank (`-`) and the `alert detail` to be something like:

```
stop codon in protein-based alignment [early_stop_occurs_at_undeterminable_mdl_position_in:<accn>.1:6247..6517:+]
```

**Comment by nawrockie** [comment]
Fixed in `develop` branch as of [090e702](https://github.com/ncbi/vadr/commit/090e702f073d71aad3f6dcb9a8df1e5c8e38ce3d)

**Comment by nawrockie** [comment]
**The #84 root cause recurs at 6 other call sites (fixed in `1b18b99`)**

While annotating an HIV-1 model, `v-annotate.pl` died during blastx
protein validation:

ERROR in vdr_CoordsRelativeSegmentToAbsolute, relative coords segment
has a position 2578 that exceeds absolute coords (6246..8813:+) length: 2568



This is the same root cause as #84: the blastx protein library contains a
**registered protein longer than the model's reference CDS** (legitimate —
HIV-1 env with insertions relative to the model reference). A blastx
**subject** coordinate is then mapped into the reference-CDS frame via
`vdr_CoordsProteinRelativeToAbsolute`, overshoots the CDS length, and dies
in `vdr_CoordsRelativeSegmentToAbsolute`.

The #84 fix added an explicit overshoot guard (→ `VADRNULL`) — but **only
at the `cdsstopp`/`trcstop` call site**. The same conversion occurs at **6
other protein-validation call sites** that were never guarded:

| call site (alert) | subject coord passed |
|---|---|
| `indfantp` model coords | `$p_hstart`/`$p_hstop` |
| `indfstrp` model coords | `$p_hstart`/`$p_hstop` |
| `insertnp` position-exception key | `$p_ins_spos` |
| `deletinp` position-exception key | `$p_del_spos` ← the one that fired |
| helper insert `mdl:` coords | subject `$aa_mpos` |
| helper delete `mdl:` coords | subject `$aa_mpos` |

**Concrete trigger:** input `FJ469726`, HIV-1 env CDS (model coords
`6246..8813:+`, 856 codons). Registered library protein `MK499378`
(878 aa) has `DEL Q436:S860-21`; subject aa 860 → nt 2578 > 2568 → fatal
on the `deletinp` path. Several other env library entries are over-length
the same way (`KT427872` 859 aa, `AF447763` 871 aa, `EF394357` 876 aa).
Note the `deletinp`/`insertnp` conversions run *unconditionally* (to build
the position-specific exception-table key) before the indel-length test, so
even a sub-threshold indel triggers the fatal.

**Fix (`1b18b99`):** factored the #84 overshoot check into a shared
predicate `vdr_CoordsProteinRelativeExceedsAbsolute()` and applied the same
`VADRNULL` convention at all 6 sites — over-length → report `VADRNULL`
model coords (or, for the indel exception-key lookups, skip the lookup and
use the default `xmaxins`/`xmaxdel`). The shared
`vdr_CoordsProteinRelativeToAbsolute` stays fatal for genuine out-of-frame
bugs. After the fix, the triggering input completes and passes.

The one remaining `vdr_CoordsProteinRelativeToAbsolute` call (the
hmmer-ORF env path) was checked and is **not** affected — its reference
frame is the ORF itself, so the protein coords are in-frame by
construction.

The same unpatched code is also present in the 1.7 release and the
`build-profile` dev branch. `1b18b99` lands the fix on `develop`; the
same patch applies cleanly to those baselines if needed.
